lib/auth/helpers.go

```diff
@@ -878,18 +879,25 @@ func NewServerIdentity(clt *Server, hostID string, role types.SystemRole) (*Iden
 		return nil, trace.Wrap(err)
 	}
 
-	keys, err := clt.GenerateServerKeys(GenerateServerKeysRequest{
-		HostID:       hostID,
-		NodeName:     hostID,
-		Roles:        types.SystemRoles{types.RoleAuth},
-		PublicTLSKey: publicTLS,
-		PublicSSHKey: pub,
-	})
+	certs, err := clt.GenerateServerKeys(context.Background(),
+		&proto.GenerateServerKeysRequest{
+			HostID:       hostID,
+			NodeName:     hostID,
+			Roles:        []string{string(types.RoleAuth)},
+			PublicTLSKey: publicTLS,
+			PublicSSHKey: pub,
+		})
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	keys.Key = priv
-	return ReadIdentityFromKeyPair(keys)
+
+	return ReadIdentityFromKeyPair(&PackedKeys{
```

Comment:
If we change the signature to `ReadIdentityFromKeyPair(privateKey []byte, certs *proto.Certs)` we could remove `auth.PackedKeys` entirely. Thoughts?

Reply:
I'm not sure I've looked at this enough to have a strong opinion. Do you mean we could remove the PackedKeys type entirely? What are we trying to achieve by doing that?

Reply:
Yes, we could remove the type entirely. Prior to this change, we had:

- GenerateUserCerts, which is in the GRPC server and returns `proto.Certs` (an SSH and TLS certificate)
- GenerateServerKeys, which did not have a GRPC endpoint but returned `PackedKeys` (an SSH and TLS certificate, the corresponding private key, as well as the CA certificates that signed them)

With this change, we update `proto.Certs` to include the CAs, making it essentially the same type as `PackedKeys`.
This is the only remaining use of `PackedKeys`, so we're converting `proto.Certs` to `PackedKeys` just so that we can call `ReadIdentityFromKeyPair`.

By removing`PackedKeys` we could reduce some potential confusion for having two similar types, and also avoid the situation we have today where `PackedKeys` usually has a nil private key field but sometimes doesn't.

Reply:
+1 for removing PackedKeys. Thanks for the explanation.